src/core/observer/index.js

```diff
@@ -218,7 +218,11 @@ export function set (obj: Array<any> | Object, key: any, val: any) {
 /**
  * Delete a property and trigger change if necessary.
  */
-export function del (obj: Object, key: string) {
+export function del (obj: Array<any> | Object, key: any) {
```

Comment:
I have tried, if I set the `key: number | string`, the type check of `obj.splice(key, 1)` would be failed. Because the first argument must be a number.